src/Pim/Bundle/CatalogBundle/EventSubscriber/IndexProductsSubscriber.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\EventSubscriber;
+
+use Akeneo\Bundle\ElasticsearchBundle\Client;
+use Akeneo\Component\StorageUtils\StorageEvents;
+use Pim\Component\Catalog\Model\ProductInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Index products in the search engine.
+ *
+ * This is not done directly in the product saver as it's only a technical
+ * problem. The product saver only handles business stuff.
+ *
+ * @author    Julien Janvier <j.janvier@gmail.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class IndexProductsSubscriber implements EventSubscriberInterface
+{
+    /** @var NormalizerInterface */
+    protected $normalizer;
+
+    /** @var Client */
+    protected $indexer;
+
+    /** @var string */
+    protected $indexType;
+
+    /**
+     * @param NormalizerInterface $normalizer
+     * @param Client              $indexer
+     * @param string              $indexType
+     */
+    public function __construct(NormalizerInterface $normalizer, Client $indexer, $indexType)
+    {
+        $this->normalizer = $normalizer;
+        $this->indexer = $indexer;
+        $this->indexType = $indexType;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            StorageEvents::POST_SAVE => 'indexProduct',
+            StorageEvents::POST_SAVE_ALL => 'bulkIndexProducts',
+        ];
+    }
+
+    /**
+     * Index one single product.
+     *
+     * @param GenericEvent $event
+     */
+    public function indexProduct(GenericEvent $event)
+    {
+        $product = $event->getSubject();
+        if (!$product instanceof ProductInterface) {
+            return;
+        }
+
+        if (!$event->hasArgument('unitary') || false === $event->getArgument('unitary')) {
+            return;
+        }
+
+        $normalizedProduct = $this->normalizer->normalize($product, 'indexing');
```

Comment:
> And it would seems logical to me that the index function takes product object as a parameter. You may need to do some stuff in the indexer that requires the product object.

The ES client I've coded in _AkeneoElasticSearchBundle_ is purely technical and business agnostic. And I've done that totally on purpose for 2 main reasons:
- easy to drop (as it's not linked to our business) if we find something more suitable (and I hope we'll find something more suitable as it's not our business / main objective to write some technical bundles)
- on the contrary, if we find nothing more suitable for our needs, that means there is maybe a miss on the Symfony scene. And then it becomes really easy (and almost ready) to opensource as it's not linked to our business. 

 > I don't think that normalizing the product should be the responsibility of the subscriber. In fact, a subscriber is a little bit like a controller: it's a glue, but shouldn't hold any technical logic.
I tend to disagree. It's a glue, so it's here to hold technical magic. 
But, nevertheless, I agree with you we could have a dedicated service to index products. It would contain these 2 lines of codes. I don't yet know if we'll need it at some other place. I suggest to introduce it when we feel we need it. Is it ok for you?